xmake/actions/run/main.lua

```diff
@@ -29,6 +29,7 @@ import("devel.debugger")
 import("async.runjobs")
 import("private.action.run.runenvs")
 import("private.service.remote_build.action", {alias = "remote_build_action"})
+import("lib.detect.check_targetname")
```

Comment:
it's private interface, please move to `private.xxx` modules.

Reply:
I am not sure how many people will use this interface in the plugin. 

At present, it is only used internally. If it is to be opened as a public interface, it must at least be mature and stable. 

As it is a newly added interface, it should be temporarily placed in the private interface, and iterated for several versions until it does not change, and then we will gradually open it.

In addition, find_similar_targetnames, as its internal implementation interface, should be moved to the private interface first.

we can move them to `private.detect.xxx` fisrt. It will not be made public until its interface name and parameters are stable after several version iterations.